lib/worker.go

```diff
@@ -67,25 +69,29 @@ func (w *LocalWorker) SetDelay(d time.Duration) {
 func (w *LocalWorker) Start(coord WorkCoordinator) error {
 	w.coord = coord
 	cfg := w.cfg
+	w.fetchers = make([]*fetchbot.Fetcher, cfg.Parallelism)
+	w.queues = make([]*fetchbot.Queue, cfg.Parallelism)
 
-	f := fetchbot.New(newMux(coord, cfg.RecordRedirects, cfg.RecordResponseHeaders))
-	f.DisablePoliteness = !cfg.Polite
-	f.CrawlDelay = time.Duration(cfg.DelayMilli) * time.Millisecond
-	f.UserAgent = cfg.UserAgent
-	if cfg.RecordRedirects {
-		f.HttpClient = NewRecordRedirectClient(coord)
-	}
-
-	w.fetcher = f
-
-	ch, err := coord.Queue()
+	ch, err := w.coord.Queue()
 	if err != nil {
 		return err
 	}
 
-	w.queue = w.fetcher.Start()
+	for i := 0; i < cfg.Parallelism; i++ {
+		f := fetchbot.New(newMux(coord, cfg.RecordRedirects, cfg.RecordResponseHeaders))
+		f.DisablePoliteness = !cfg.Polite
+		f.CrawlDelay = time.Duration(cfg.DelayMilli) * time.Millisecond
+		f.UserAgent = cfg.UserAgent
+		if cfg.RecordRedirects {
+			f.HttpClient = NewRecordRedirectClient(coord)
+		}
+
+		w.fetchers[i] = f
+		w.queues[i] = f.Start()
+	}
 
 	go func() {
+		i := 0
```

Comment:
This isn’t different from the previous version, but it just occurred to me that this implementation will just automatically consume everything from the coordinator’s queue and put it in an internal queue here. Since the workers config is an array, don’t we need this to only consume from the coordinator’s queue when a fetchbot is free?

Reply:
Thinking more about this:
- If one URL is especially slow to load, other URLs will get stuck behind it in the queue for the fetchbot handling it.
- Should the workers config even be an array? I'm trying to think of a situation where we’d *want* a heterogenous set of scraping algorithms (except where they *all* work every item in the queue, which is not how this component is set up to work).

Reply:
oh truth. This is interesting. I vote we work on this in another PR.

Reply:
>> …Since the workers config is an array, don’t we need this to only consume from the coordinator’s queue when a fetchbot is free?
>
> …I vote we work on this in another PR.

👍